itou/users/management/commands/pe_certify_users.py

```diff
@@ -0,0 +1,102 @@
+            User.objects.filter(
+                kind=UserKind.JOB_SEEKER,
+                is_active=True,
+                jobseeker_profile__isnull=False,  # normally these days everyone should have one. Should we keep this?
+                jobseeker_profile__pe_certified_at=None,  # only those never seen yet. Later we can introduce a retry.
+                jobseeker_profile__id_national_pe=None,
+            )
+            .select_related("jobseeker_profile")
+            .order_by("-pk")
+        )  # most recent users first, they are the top priority.
+        self.stdout.write(f"> about to resolve first_name and last_name for count={active_job_seekers.count()} users.")
+
+        eligible_users = active_job_seekers.exclude(Q(nir="") | Q(birthdate=None) | Q(first_name="") | Q(last_name=""))
+        self.stdout.write(f"> only count={eligible_users.count()} users have the necessary data to be resolved.")
+
+        examined_profiles = []
+        certified_profiles = []
+        swapped_users = []
+
+        def certify_user(user, id_certifie):
+            user.jobseeker_profile.id_national_pe = id_certifie
+            certified_profiles.append(user.jobseeker_profile)
+            self.stdout.write(f"> certified user pk={user.pk} id_certifie={id_certifie}")
+
+        for user in eligible_users[:chunk_size]:
+            user.jobseeker_profile.pe_certified_at = timezone.now()
+            examined_profiles.append(user.jobseeker_profile)
+            response = pe_check_user_details(user)
+            if isinstance(response, Exception):
+                self.stdout.write(
+                    f"! could not find a match for pk={user.pk} first_name={user.first_name} "
+                    f"last_name={user.last_name} birthdate={user.birthdate} nir={user.nir} error={response}"
+                )
+                response2 = pe_check_user_details(user, swap=True)
+                if isinstance(response2, Exception):
+                    self.stdout.write(
+                        f"! no match found either for first_name={user.last_name} last_name={user.first_name} "
+                        f"exc={response2}"
+                    )
+                else:
+                    self.stdout.write(f"> SWAP DETECTED ! user pk={user.pk} id_certifie={response2}")
+                    user.last_name, user.first_name = user.first_name, user.last_name
+                    certify_user(user, response2)
+                    swapped_users.append(user)
+            else:
+                certify_user(user, response)
+
+        if wet_run:
+            JobSeekerProfile.objects.bulk_update(examined_profiles, ["pe_certified_at"], batch_size=1000)
+            self.stdout.write(f"> count={len(examined_profiles)} users have been examined.")
+            JobSeekerProfile.objects.bulk_update(certified_profiles, ["id_national_pe"], batch_size=1000)
+            self.stdout.write(f"> count={len(certified_profiles)} users have been certified.")
```

Comment:
```suggestion
            self.stdout.write(f"> count={len(examined_profiles)} users have been examined.")
            JobSeekerProfile.objects.bulk_update(certified_profiles, ["pe_certified_at", "id_national_pe"], batch_size=1000)
            self.stdout.write(f"> count={len(certified_profiles)} users have been certified.")
            not_certified = set(examined_profiles) - set(certified_profiles)
            JobSeekerProfile.objects.bulk_update(not_certified, ["pe_certified_at"], batch_size=1000)
            self.stdout.write(f"> count={len(not_certified)} users could not be certified.")
```
(et cette ligne `JobSeekerProfile.objects.bulk_update(not_certified, ["pe_certified_at"], batch_size=1000)` confirme que le nom `pe_certified_at` n'est pas idéal :grimacing: )

Reply:
hehe, traité.